### PR TITLE
[V2] Add Protocol Property

### PR DESCRIPTION
**Pull request by shbatm**
Added a `protocol` property to each node, group, program, and variable to determine the root protocol used:
- Insteon
- Z-Wave
- Zigbee
- Node Server (w/ Slot #)
- Program
- Program Folder
- State/Integer Variable

Capture the `family` protocol for nodes and groups as per the ISY-WSDK-5.0.4:
```
0 = Default (core driver implementation e.g. Insteon, UPB)
1 = Insteon products
2 = UPB Products
3 = RCS Products
4 = ZWave products
5 = Auto DR (for groups)
6 = Generic (for groups)
7 = UDI Products
8 = Brultech Products
9 = NCD Products
```

Add more string literals and cleanup existing constants to be more descriptive.

### CODE REVIEW

**Comment by shbatm** [comment]
@OverloadUT I ended up creating a project board on my shbatm/hacs-isy994 page, just to track what was outstanding between this V2 branch and PyISY-beta, (and to start planning how to get all of the changes merged into Home Assistant after V2 release).

For reference (and sorry for flooding you w PRs, just trying to keep them small for ease of review):
https://github.com/shbatm/hacs-isy994/projects/1

File: PyISY/Programs/program.py [hunk 1]
```diff
@@ -104,3 +104,8 @@ def update(self, wait_time=0, data=None):
                 self.running.update(prunning, force=True, silent=True)
             elif not self.isy.auto_update:
                 self._programs.update(wait_time, pid=self._id)
+
+    @property
+    def protocol(self):
+        """Return the protocol for this entity."""
+        return "program"
```
Comment (OverloadUT) [inline PyISY/Programs/program.py]: make const

File: PyISY/Variables/variable.py [hunk 2]
```diff
@@ -75,6 +75,11 @@ def nid(self):
         """Return the formatted Variable Type and ID."""
         return "{!s}.{!s}".format(self._type, self._id)
 
+    @property
+    def protocol(self):
+        """Return the protocol for this entity."""
+        return "{} variable".format("integer" if self._type == "1" else "state")
```
Comment (OverloadUT) [inline PyISY/Variables/variable.py]: make consts

**Comment by shbatm** [comment]
May have gotten a bit carried away with the string literals, but I got on a roll so I went with it. Let me know if you see any issues.

After looking through the `ISY-WSDK-5.0.4` documentation, I added the family property as well as the protocol we're adding (they don't directly translate 1-to-1).

**Pull request merged**